mlflow/sklearn/__init__.py

```diff
@@ -932,9 +1390,46 @@ def patched_fit(original, self, *args, **kwargs):
         )
     ]
 
+    def should_patch_class_method(class_def, func_name):
```

Comment:
Can we add a docstring to this method and explain why we need it? I.e. because certain scikit-learn model classes define training APIs as properties, rather than callable methods?